frontends/concrete-python/examples/levenshtein_distance/levenshtein_distance.py

```diff
@@ -225,50 +229,45 @@ def _compute_in_fhe(self, list_patterns: list, show_distance: bool = False):
 @fhe.module()
 class LevenshsteinModule:
     @fhe.function({"x": "encrypted", "y": "encrypted"})
-    def equal(x, y):
-        """Assert equality between two chars of the alphabet."""
-        return x == y
+    def char_cost_align(x, y):
+        """Cost of having two chars of the alphabet aligned."""
+        return x != y
 
     @fhe.function({"x": "clear"})
     def constant(x):
         return fhe.zero() + x
 
     @fhe.function(
         {
-            "is_equal": "encrypted",
-            "if_equal": "encrypted",
-            "case_1": "encrypted",
-            "case_2": "encrypted",
-            "case_3": "encrypted",
+            "char_cost_align": "encrypted",
+            "align": "encrypted",
+            "insertion_1": "encrypted",
+            "insertion_2": "encrypted",
         }
     )
-    def mix(is_equal, if_equal, case_1, case_2, case_3):
-        """Compute the min of (case_1, case_2, case_3), and then return `if_equal` if `is_equal` is
-        True, or the min in the other case."""
-        min_12 = np.minimum(case_1, case_2)
-        min_123 = np.minimum(min_12, case_3)
-
-        return fhe.if_then_else(is_equal, if_equal, 1 + min_123)
-
-    # There is a single output in mix: it can go to
-    #   - input 1 of mix
-    #   - input 2 of mix
-    #   - input 3 of mix
-    #   - input 4 of mix
+    def min_cost(char_cost_align, align, insertion_1, insertion_2):
+        """Compute `min(align + char_cost_align, insertion_1 + 1, insertion_2 + 1)`."""
+        align = align + char_cost_align
+        insertion_min = 1 + np.minimum(insertion_1, insertion_2)
+        result = np.minimum(align, insertion_min)
+        return fhe.refresh(result) # to have at least one operation reducing the noise
```

Comment:
It seems this PR changes the implementation of levenshtein distance example, if there is any other material related to it (e.g., docs) they should be updated as well.

Reply:
yes it was used as test code. this draft lacks real tests. I am cleaning.

Reply:
No, I mean explanation of the algorithm, like as a tutorial. Like https://github.com/zama-ai/concrete/blob/main/frontends/concrete-python/examples/levenshtein_distance/README.md. 

If an example changes, we need to change the tutorial that explains is as well.

Reply:
Removed from PR